### ISSUE TITLE
provider.azurerm v1.22.0 crashes terraform v0.11.11 and causes error locking tfstate

### ISSUE THREAD

**Issue by enorlando** [terraform-providers/terraform-provider-azuread#39]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

Terraform v0.11.11
+ provider.azuread v0.1.0
+ provider.azurerm v1.22.0

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Panic Output

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x176c6a8]

goroutine 6110 [running]:
github.com/hashicorp/terraform/terraform.(*NodeRefreshableManagedResourceInstance).evalTreeManagedResourceNoState(0xc000308030, 0xa2176d0, 0xc0008480a0)
	/private/tmp/terraform-20181217-80581-zpqwjg/terraform-0.11.11/src/github.com/hashicorp/terraform/terraform/node_resource_refresh.go:238 +0x328
github.com/hashicorp/terraform/terraform.(*NodeRefreshableManagedResourceInstance).EvalTree(0xc000308030, 0x3ae1700, 0xc000308030)
	/private/tmp/terraform-20181217-80581-zpqwjg/terraform-0.11.11/src/github.com/hashicorp/terraform/terraform/node_resource_refresh.go:98 +0x6e
github.com/hashicorp/terraform/terraform.(*Graph).walk.func1(0x3ae1700, 0xc000308030, 0x0, 0x0)
	/private/tmp/terraform-20181217-80581-zpqwjg/terraform-0.11.11/src/github.com/hashicorp/terraform/terraform/graph.go:113 +0x9a2
github.com/hashicorp/terraform/dag.(*Walker).walkVertex(0xc0009f8000, 0x3ae1700, 0xc000308030, 0xc0002fe2c0)
	/private/tmp/terraform-20181217-80581-zpqwjg/terraform-0.11.11/src/github.com/hashicorp/terraform/dag/walk.go:387 +0x367
created by github.com/hashicorp/terraform/dag.(*Walker).Update
	/private/tmp/terraform-20181217-80581-zpqwjg/terraform-0.11.11/src/github.com/hashicorp/terraform/dag/walk.go:310 +0x986

### Expected Behavior

Terraform should have successfully run a plan.

### Steps to Reproduce

1.  az login
2.  terraform init
3.  terraform plan -->causes panic
4. terraform plan -->causes error locking state

### Important Factoids

Below error when running terraform plan second time. When the tfstate state is leased in azure, it will lock again running a plan

Error: Error locking state: Error acquiring the state lock: storage: service returned error: StatusCode=409, ErrorCode=LeaseAlreadyPresent, ErrorMessage=There is already a lease present.
RequestId:b2592325-701e-004a-607e-c2a9a8000000
Time:2019-02-12T02:57:14.9888560Z, RequestInitiated=Tue, 12 Feb 2019 02:57:14 GMT, RequestId=b2592325-701e-004a-607e-c2a9a8000000, API Version=2016-05-31, QueryParameterName=, QueryParameterValue=
Lock Info:
  ID:        xxxxx
  Path:      xxxxxx
  Operation: OperationTypePlan
  Who:       xxxxxx
  Version:   0.11.11
  Created:   2019-02-12 02:56:55.201233 +0000 UTC
  Info:

Terraform acquires a state lock to protect the state from being written
by multiple users at the same time. Please resolve the issue above and try
again. For most commands, you can disable locking with the "-lock=false"
flag, but this is not recommended.

* #0000

**Comment by katbyte** [comment]
Hi @enorlando

It looks like something in your configuration is causing terraform to panic, would you mind sharing with us some terraform config that triggers this crash so we can see whats going on? Thanks!

**Comment by enorlando** [comment]
Hi @katbyte 

I'm not sure where in the terraform config it is crashing. I have enabled trace logging and searched through the logs however it does not show me where it crashes during the plan. How can I pinpoint when it panic's as we have many terraform files for different azure configurations?

**Comment by tombuildsstuff** [comment]
hi @enorlando 

Taking a look into this - this error appears to be coming from a bug in the AzureRM Backend in Terraform Core, rather than something specific to this Provider. This bug exists in the Backend within Terraform 0.11 and was being tracked [in this issue](https://github.com/hashicorp/terraform/issues/17046) - however [a fix for this has been merged](https://github.com/hashicorp/terraform/pull/19441) and will ship as a part of Terraform 0.12.

At this time it should be possible to unlock the Blob File within the Storage Account and re-try again, which should give you the underlying error (which should tell us how to proceed with this) - would you be able to take a look?

Thanks!

**Comment by enorlando** [comment]
Hi @tombuildsstuff 

This is the error received when I break the lease from the Blob file and re-try Terraform Plan

Error: Error locking state: Error acquiring the state lock: storage: service returned error: StatusCode=409, ErrorCode=LeaseAlreadyPresent, ErrorMessage=There is already a lease present.
RequestId:22e11e46-f01e-005b-4c1d-c79eb3000000
Time:2019-02-18T00:04:17.2246370Z, RequestInitiated=Mon, 18 Feb 2019 00:04:16 GMT, RequestId=22e11e46-f01e-005b-4c1d-c79eb3000000, API Version=2016-05-31, QueryParameterName=, QueryParameterValue=
Lock Info:
  ID:        <>
  Path:      terraform/<>.terraform.tfstate
  Operation: OperationTypePlan
  Who:       <>
  Version:   0.11.11
  Created:   2019-02-14 21:52:51.034163 +0000 UTC
  Info:      


Terraform acquires a state lock to protect the state from being written
by multiple users at the same time. Please resolve the issue above and try
again. For most commands, you can disable locking with the "-lock=false"
flag, but this is not recommended.

**Comment by enorlando** [comment]
Hi @tombuildsstuff 
Have you had a chance to look at the issue? What is the ETA for Terraform 0.12 release?
Thanks

**Comment by tombuildsstuff** [comment]
hi @enorlando 

Thanks for confirming that - this is a duplicate of https://github.com/hashicorp/terraform/issues/17046  and as such I'm going to close it in favour of that issue, which has been resolved and the fix is available as of Terraform 0.12.

Regarding a release date for Terraform 0.12 - [Beta 1 of Terraform 0.12 has recently been released](https://www.hashicorp.com/blog/announcing-terraform-0-1-2-beta1) however there's no more information about the release date available at this time, unfortunately.

Thanks!
